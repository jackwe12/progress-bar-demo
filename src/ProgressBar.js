import React from 'react';
// import PropTypes from 'prop-types';
import ProgressBar from 'react-bootstrap/ProgressBar';

const Bar = ({barValue, limit}) =>{

    return (
        (barValue < limit)
        ?(
            <div className='bar'>
                <ProgressBar 
                now={barValue}
                label={`${barValue}%`}    
                />
            </div>
        )
        :(
            <div className='bar'>
                <ProgressBar 
                now={barValue}
                label={`${barValue}%`}    
                variant='danger'
                />
            </div>
        )
    );

}

// ProgressBar.propTypes = {
//     barValue: PropTypes.array.isRequired,
//     limit: PropTypes.number.isRequired
// }


export default Bar;