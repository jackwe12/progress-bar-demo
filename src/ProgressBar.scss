.progress{
    width:50%;
    height:3rem!important;
    margin-bottom:0.5rem;
    // justify-content: center;
    font-size: 1rem!important;
    position:relative;

}
.progressBar{
    display:flex;
    flex-direction: column;
    background-color: #007bff;
}
.bar-value{
    position: absolute;
    top: 25%;
    left: 50%;
    bottom: 0;
    right: 0;
    overflow: hidden;
}
.danger{
    background-color: #dc3545!important;
}
// .progress-bar{
//     position: absolute;
//     top: 0;
//     left: 0;
//     bottom: 0;
//     right: 0;
//     z-index: -1;
//     overflow: hidden;
// }