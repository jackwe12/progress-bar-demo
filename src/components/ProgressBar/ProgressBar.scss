.progress{
    width:50%;
    height:3rem!important;
    margin-bottom:1rem!important;
    font-size: 1rem!important;
    position:relative;

}
.progressBar{
    display:flex;
    flex-direction: column;
    background-color: #007bff;
    transition: width 0.5s;

}
.bar-value{
    position: absolute;
    top: 25%;
    left: 50%;
    bottom: 0;
    right: 0;
    overflow: hidden;
}
.danger{
    background-color: #dc3545!important;
}
@media screen and (max-width:500px){
    .progress{
        width:80%;
    }
    
}