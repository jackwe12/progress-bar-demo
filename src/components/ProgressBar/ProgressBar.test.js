import ProgressBar from './ProgressBar';
import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { shallow} from 'enzyme';

configure({ adapter: new Adapter() });

const wrapper = shallow(
    <ProgressBar
        barValue = {100}
        limit = {120}
        key = {0}
    />);
const wrapper2 = shallow(
    <ProgressBar
        barValue = {130}
        limit = {120}
        key = {0}
    />);

describe('Testing <ProgressBar/>', ()=>{
    test('snapshot testing',()=>{
        const tree =renderer
            .create(<ProgressBar
                    barValue = {100}
                    limit = {120}
                    key = {0}
             />)
            .toJSON();
            expect(tree).toMatchSnapshot();
    });

    test('buttonValue is right',()=>{
        expect(wrapper.find('.bar-value').text()).toEqual('100%')
    });

    test('buttonValue is larger than limit, turn danger',()=>{
        let bar = wrapper2.find('.progressBar');
        expect(bar.hasClass('danger')).toEqual(true);
    });


});
