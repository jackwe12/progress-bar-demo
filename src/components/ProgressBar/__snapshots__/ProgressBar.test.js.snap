// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <ProgressBar/> snapshot testing 1`] = `
<div
  className="bar"
>
  <div
    className="progress"
  >
    <div
      className="bar-value"
    >
      100
      %
    </div>
    <div
      className="progressBar"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;
